packages/frontend-api/src/Model/Resolver/Price/SpecialPriceApiFactory.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Resolver\Price;
+
+use DateTimeImmutable;
+use Shopsys\FrameworkBundle\Model\Pricing\Price;
+use Shopsys\FrameworkBundle\Model\Pricing\SpecialPrice\SpecialPrice;
+use Shopsys\FrameworkBundle\Model\Pricing\SpecialPrice\SpecialPriceFactory;
+use Shopsys\FrameworkBundle\Model\Product\Pricing\PriceFactory;
+
+class SpecialPriceApiFactory
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\SpecialPrice\SpecialPriceFactory $specialPriceFactory
+     * @param \Shopsys\FrameworkBundle\Model\Product\Pricing\PriceFactory $priceFactory
+     */
+    public function __construct(
+        protected readonly SpecialPriceFactory $specialPriceFactory,
+        protected readonly PriceFactory $priceFactory,
+    ) {
+    }
+
+    /**
+     * @param array $data
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Price $basicPrice
+     * @return \Shopsys\FrameworkBundle\Model\Pricing\SpecialPrice\SpecialPrice|null
+     */
+    public function createSpecialPriceFromArray(array $data, Price $basicPrice): ?SpecialPrice
+    {
+        $specialPricesArray = $data['special_prices'];
+
+        if (count($specialPricesArray) === 0) {
+            return null;
+        }
+
+        $effectiveSpecialPrice = $this->getEffectiveSpecialPrice($specialPricesArray);
+
+        if ($effectiveSpecialPrice === null || $effectiveSpecialPrice->price->getPriceWithVat()->isLessThan($basicPrice->getPriceWithVat())) {
+            return $effectiveSpecialPrice;
+        }
+
+        return null;
+    }
```

Comment:
I assume the returns should be inverted too.

Maybe this could work too 

```php
if ($effectiveSpecialPrice?->price->getPriceWithVat()->isGreaterThanOrEqualTo($basicPrice->getPriceWithVat())) {
    return null;
}

return $effectiveSpecialPrice
```

Reply:
98e64a83a9